peekingduck/pipeline/nodes/model/jde_mot/jde_files/multitracker.py

```diff
@@ -0,0 +1,546 @@
+
+        self.score = new_track.score
+        if update_feature:
+            self.update_features(new_track.curr_feat)
+
+    @property
+    def tlwh(self) -> np.ndarray:
+        """Get current position in bounding box format.
+
+        Returns:
+            np.ndarray: `(top left x, top left y, width, height)`.
+        """
+        if self.mean is None:
+            return self._tlwh.copy()
+        ret = self.mean[:4].copy()
+        ret[2] *= ret[3]
+        ret[:2] -= ret[2:] / 2
+        return ret
+
+    @property
+    def tlbr(self) -> np.ndarray:
+        """Convert bounding box to format `(min x, min y, max x, max y)`.
+
+        Returns:
+            np.ndarray: `(top left, bottom right)`.
+        """
+        ret = self.tlwh.copy()
+        ret[2:] += ret[:2]
+        return ret
+
+    @staticmethod
+    def tlwh_to_xyah(tlwh: torch.Tensor) -> np.ndarray:
+        """Convert bounding box to format `(center x, center y, aspect ratio,
+        height)`, where the aspect ratio is `width / height`.
+
+        Args:
+            tlwh (torch.Tensor): Bounding box with format
+                `(top left x, top left y, width, height)`.
+
+        Returns:
+            np.ndarray: `(center x, center y, aspect ratio, height)`
+        """
+        ret = np.asarray(tlwh).copy()
+        ret[:2] += ret[2:] / 2
+        ret[2] /= ret[3]
+        return ret
+
+    def to_xyah(self) -> np.ndarray:
+        """Convert bounding box format from (top left x, top left y, width,
+        height) to (center x, center y, aspect ratio, height)."""
+        return self.tlwh_to_xyah(self.tlwh)
+
+    @staticmethod
+    def tlbr_to_tlwh(tlbr: torch.Tensor) -> np.ndarray:
+        """Convert format of bounding box.
+
+        Args:
+            tlbr (torch.Tensor): `(min x, min y, max x, max y)`.
+
+        Returns:
+            np.ndarray: `(top left x, top left y, width, height)`.
+        """
+        ret = np.asarray(tlbr).copy()
+        ret[2:] -= ret[:2]
+        return ret
+
+    @staticmethod
+    def tlwh_to_tlbr(tlwh: torch.Tensor) -> np.ndarray:
+        """Convert format of bounding box.
+
+        Args:
+            tlwh (torch.Tensor): `(top left x, top left y, width, height)`.
+
+        Returns:
+            np.ndarray: `(min x, min y, max x, max y)`.
+        """
+        ret = np.asarray(tlwh).copy()
+        ret[2:] += ret[:2]
+        return ret
+
+    def __repr__(self) -> str:
+        return f"OT_{self.track_id}_({self.start_frame}-{self.end_frame})"
+
+
+class JDETracker(object):
+    """JDE Tracker class"""
+
+    def __init__(
+        self, opt: Dict[str, Any], device: torch.device, frame_rate: int = 30
+    ) -> None:
+        self.opt = opt
+        self.model = Darknet(opt["cfg"], nID=14455)
+        self.model.load_state_dict(
+            torch.load(opt["weights_path"], map_location="cpu")["model"], strict=False
+        )
+        self.model.eval().to(device)
+
+        self.tracked_stracks = []  # type: List[STrack]
```

Comment:
prefer `self.tracked_stracks: List[STrack] = []` instead of inline comment to allow syntax highlighting

similar for below

Reply:
Updated accordingly